carbon-p2-plugin/src/main/java/org/wso2/maven/p2/utils/FileManagementUtil.java

```
@@ -0,0 +1,291 @@
+* under the License.
+*/
+package org.wso2.maven.p2.utils;
+
+import org.apache.maven.plugin.logging.Log;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Properties;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+import java.util.zip.ZipOutputStream;
+
+/**
+ * Util class which handle file manipulation operations.
+ */
+public class FileManagementUtil {
+    private static final int BUFFER = 2048;
+
+
+    /**
+     * For a given location and the profile, returns the File object represented by config.ini
+     *
+     * @param destination path pointing the [carbon product]/repository/components folder
+     * @param profile     name of the profile
+     * @return the config.ini file for the Profile
+     */
+    public static File getProfileConfigIniFile(String destination, String profile) {
+        return new File(destination + File.separator + profile + File.separator + "configuration" +
+                File.separator + "config.ini");
+    }
+
+    /**
+     * Updated the given config.ini file with a given key value pair.
+     *
+     * @param configIniFile File object representing the config.ini
+     * @param propKey       property key
+     * @param value         property value
+     */
+    public static void changeConfigIniProperty(File configIniFile, String propKey, String value, Log log) {
+        Properties prop = new Properties();
+
+        try (InputStream inputStream = new FileInputStream(configIniFile)) {
+            prop.load(inputStream);
+            prop.setProperty(propKey, value);
+            try (OutputStream outputStream = new FileOutputStream(configIniFile)) {
+                prop.store(outputStream, null);
+            }
+        } catch (IOException e) {
+            log.warn(e);
```

Shouldn't we propagate the error back to the caller? If the property update was not successful, we need to inform the caller IMO.